File: src/main/java/io/lsdconsulting/lsd/distributed/generator/diagram/event/builder/ConsumeMessageBuilder.java
package io.lsdconsulting.lsd.distributed.generator.diagram.event.builder;

import com.lsd.core.IdGenerator;
import com.lsd.core.domain.SequenceEvent;
import io.lsdconsulting.lsd.distributed.generator.diagram.event.SequenceEventBuilder;
import org.springframework.stereotype.Component;

import static com.lsd.core.builders.MessageBuilder.messageBuilder;
import static com.lsd.core.domain.MessageType.SYNCHRONOUS;

@Component
public class ConsumeMessageBuilder implements SequenceEventBuilder {
    @Override
    public SequenceEvent build(IdGenerator idGenerator, String label, String serviceName, String target, String colour, String data) {
        return messageBuilder()
                .id(idGenerator.next())
                .type(SYNCHRONOUS)
                .label(label)
                .from(target)
                .to(serviceName)
                .colour(colour)
                .data(data)
                .build();
    }
}

File: src/main/kotlin/io/lsdconsulting/lsd/distributed/generator/diagram/event/builder/ConsumeMessageBuilder.kt
package io.lsdconsulting.lsd.distributed.generator.diagram.event.builder

import com.lsd.core.builders.MessageBuilder.Companion.messageBuilder
import com.lsd.core.domain.Message
import com.lsd.core.domain.MessageType
import io.lsdconsulting.lsd.distributed.generator.diagram.event.MessageData

fun buildConsumeMessage(request: MessageData): Message =
    with(request) {
        messageBuilder()
            .id(id)
            .type(MessageType.SYNCHRONOUS)
            .label(captured.label)
            .from(captured.target)
            .to(captured.serviceName)
            .colour(captured.colour)
            .data(captured.data)
            .duration(captured.duration)
            .build()
    }
